tests/cpp/test_matmul.cpp

```diff
@@ -3706,7 +3708,26 @@ TEST_F(HopperMatmulTest, HSH_NT_128BSwizzle) {
 
   auto tv3 = castOp(DataType::Half, tv2);
 
-  fusion.addOutput(tv3);
+  TensorView* tv4 = nullptr;
+  if (use_smem_epilogue) {
+    // Copy from shared memory to global using TMA
+    tv4 = set(tv3);
+
+    tv3->setMemoryType(MemoryType::Shared);
+    tv4->definition()->as<LoadStoreOp>()->setOpType(
+        LoadStoreOpType::CpAsyncBulkTensorTile);
+
+    fusion.addOutput(tv4);
+
+    // We'll use stmatrix.x4 to store from reg to shared memory
+    fusion.manage("st_matrix_m_tile", 16);
```

Comment:
Just to be clear, does this mean we must use a macro such that getN(macro) is a multiple of 16?

Reply:
yes -
For the m-dimension this shouldn't be a problem since it's always 64 for Hopper.
For the n-dimension, if it's not a multiple of 16, it has to a multiple of 8, and we'll have to use stmatrix.x2 where st_matrix_n_tile will be 8.